lib/run-compilation.js

```diff
@@ -41,12 +40,12 @@ module.exports = function runCompilation (options, compiler, done) {
 
   compiler.plugin('after-emit', function afterEmit (compilation, callback) {
     if (warnings.length) {
-      compilation.warnings.push(chalk.yellow(options.formatter(warnings)));
+      compilation.warnings.push(new Error(options.formatter(warnings)));
```

Comment:
Interesting, good to know. And this behaviour is the same across v1, v2, and v3?